### PR TITLE
Added WSN x-kaazing-idle-timeout k3po test scenarios

### PR DESCRIPTION
**Pull request by irina-mitrea-luxoft**

### CODE REVIEW

File: transport/wsn/pom.xml [hunk 1]
```diff
@@ -150,6 +150,12 @@
             <version>2.0.1</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.kaazing</groupId>
+            <artifactId>specification.x-kaazing-idle-timeout</artifactId>
+            <version>develop-SNAPSHOT</version>
```
Comment (irina-mitrea-luxoft) [inline transport/wsn/pom.xml]: TODO: To be changed to the actual released version after this project has been released.

**Comment by cmebarrow** [comment]
@irina-mitrea-luxoft I would recommend pulling in the latest changes from upstream/develop, your branch is quite far behind now

File: transport/wsn/src/test/java/org/kaazing/gateway/transport/wsn/specification/extensions/idletimeout/IdleTimeoutExtensionWSNAcceptorIT.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        k3po.finish();
+    }
+
+    @Test
+    @Specification("extension.pong.frames.sent.by.server.no.client.timeout/request")
+    public void extensionPongFramesSentByServerNoClientTimeout() throws Exception {
+        acceptor.bind("ws://localhost:8001/echo", new IoHandlerAdapter<IoSessionEx>());
+        k3po.finish();
+    }
+}
```
Comment (cmebarrow) [inline transport/wsn/src/test/java/org/kaazing/gateway/transport/wsn/specification/extensions/idletimeout/IdleTimeoutExtensionWSNAcceptorIT.java]: Missing EOL character

File: transport/wsn/src/test/java/org/kaazing/gateway/transport/wsn/specification/extensions/idletimeout/IdleTimeoutExtensionWSNAcceptorIT.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+    private final WsnAcceptorRule acceptor = new WsnAcceptorRule();
+
+    @Rule
+    public TestRule chain = createRuleChain(acceptor, k3po);
+
+    @Test
+    @Specification("standard.pong.frames.sent.by.server.no.client.timeout/request")
+    public void standardPongFramesSentByServerNoClientTimeout() throws Exception {
+        acceptor.bind("ws://localhost:8001/echo", new IoHandlerAdapter<IoSessionEx>());
```
Comment (cmebarrow) [inline transport/wsn/src/test/java/org/kaazing/gateway/transport/wsn/specification/extensions/idletimeout/IdleTimeoutExtensionWSNAcceptorIT.java]: You'll need to make the handler send data every tick seconds (as in the corresponding response.rpt script). You can override method "sessionOpened" to do that, and use session.write to write the data. You'll need to create an IoBufferEx object to contain the data, using the buffer allocator you can get from the session.

**Comment by irina-mitrea-luxoft** [comment]
@cmebarrow I have performed the gateway fixes on the x-kaazing-idle-timeout WSN acceptor scenarios.

I have had a sporadic failure issue with one of the WSN Acceptor test cases, the failure was very difficult to encounter (I think one failed attempt of maybe 30-50 successful attempts). I am attaching the k3po debug log.

[201510121905_standardPongFramesSentByServerNoClientTimeout.log | uploaded via ZenHub](https://files.zenhub.io/561bdfb1592eadf906d5fb7f)

**Pull request merged**